code/turf/floors.dm

```diff
@@ -2677,6 +2697,109 @@ TYPEINFO(/turf/simulated/floor/auto/water/ice)
 		if(prob(10))
 			src.icon_state = "snow_rough[rand(1,3)]"
 
+TYPEINFO(/turf/simulated/floor/auto/glassblock)
+	mat_appearances_to_ignore = list("steel","synthrubber","glass")
+	connect_diagonal = 1
+
+//dont use please
```

Comment:
Then abstract it?

Reply:
already tried, every map check proceeded to fail

Reply:
/turf/simulated/floor is a used type